src/IoC/InstancePool.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace IoC;
+
+public sealed class InstancePool
+{
+    //This is where the instances will be stored.
+    private static Dictionary<string, object>? _implementationStore;
```

Comment:
Why are we adding question mark here?

Reply:
Shouldn't this be readonly?

Reply:
> Why are we adding question mark here?

The project is using a `nullable check` that makes possible to set to null a variable or instance. That avoid many crashes if null is received.

Reply:
> Shouldn't this be readonly?

That's a very good idea. I'll update that part and provide the result shortly.